test/object-store/sync/app.cpp

```diff
@@ -2062,7 +2241,7 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         // This assumes that we make an http request for the new token while
         // already in the WaitingForAccessToken state.
         bool seen_waiting_for_access_token = false;
-        transport->request_hook = [&](Request&) {
+        transport->request_hook = [&](Request&) -> bool {
```

Comment:
you don't have to change this, but if a lambda returns a single type - i.e. you just add a return true to the end - you don't need to make the return type explicit.